task/run-opm-command-oci-ta/0.1/run-opm-command-oci-ta.yaml

```diff
@@ -0,0 +1,309 @@
+          cpu: 500m
+          memory: 2Gi
+    - name: replace-related-images-pullspec-in-catalog
+      image: quay.io/konflux-ci/konflux-test@sha256:3619ffde751d337d02f1f61c83c5c39eb87d8091dbe5a9af58ea98577fa09461
+      workingDir: /var/workdir/source
+      securityContext:
+        runAsUser: 0
+      when:
+        - input: "$(params.IMDS_PATH)"
+          operator: notin
+          values: ["", "null"]
+      command: ["python3", "-c"]
+      args:
+        - |
+          import json
+          import os
+          import sys
+          import re
+          import yaml
+
+          def load_data_to_dict(file_path: str) -> tuple[list, str]:
+              """
+              Loads data from a specified file into a Python list of dictionaries.
+
+              Args:
+                  file_path (str): The path to the data file (e.g., YAML or JSON).
+
+              Returns:
+                  list: A list containing the loaded data as dictionaries.
+                        Each element in the list corresponds to a document in the file
+                        (e.g., a single YAML document or a single JSON object,
+                        or multiple JSON objects if concatenated).
+
+              Raises:
+                  FileNotFoundError: If the specified file path does not exist.
+                  ValueError: If the file content cannot be successfully loaded as either
+                              YAML or JSON.
+              """
+              if not file_path:
+                  return [], ''
+
+              if not os.path.exists(file_path):
+                  raise FileNotFoundError(f"File at path '{file_path}' was not found.")
+
+              with open(file_path, 'r', encoding='utf-8') as f:
+                  try:
+                      data = list(yaml.safe_load_all(f))
+                      return [doc for doc in data if doc is not None], 'yaml'
+                  except yaml.YAMLError:
+                      f.seek(0)
+                      try:
+                          return [json.load(f)], 'json'
+                      except json.JSONDecodeError:
+                          f.seek(0)
+                          result = []
+                          for part in re.split(r'(?<=})\n(?={)', f.read()):
+                              try:
+                                  result.append(json.loads(part))
+                              except json.JSONDecodeError as e:
+                                  raise ValueError(
+                                      f"File content from '{file_path}' contains malformed JSON parts "
+                                      f"after initial JSON load attempt failed: {e}")
+                          return result, 'json'
+                      except Exception as e:
+                          raise ValueError(
+                              f"An unexpected error occurred during JSON loading from '{file_path}': {e}")
+
+
+          def parse_image_digest_mirror_set(imds_data: list) -> dict:
```

Comment:
Consider using an existing function in konflux-test instead of writing a new one? 
https://github.com/konflux-ci/konflux-test/blob/main/test/utils.sh#L815

Reply:
That is good question, I probably missed functions already present. 
On the other hand, should we accept only YAML files. My impression was that ImageDigestMirrorSet can be defined in both formats JSON or YAML.

Reply:
If we want to support additional functionality, we should support it in the common function.